superset/datasets/dao.py

```diff
@@ -178,33 +178,29 @@ def update_columns(
         """
 
         column_by_id = {column.id: column for column in model.columns}
-        columns = []
 
         for properties in property_columns:
             if "id" in properties:
-                columns.append(
```

Comment:
This was the root of the problem, i.e., existing columns shouldn't be added to the set of columns which need to be bound to the dataset when updated.